velox/exec/VectorHasher.cpp

```diff
@@ -247,36 +246,43 @@ bool VectorHasher::makeValueIdsDecoded(
   auto values = decoded_.data<T>();
 
   bool success = true;
-  rows.applyToSelected([&](vector_size_t row) INLINE_LAMBDA {
+  int decodedBaseVisited = 0;
+  rows.testSelected([&](vector_size_t row) {
     if constexpr (mayHaveNulls) {
       if (decoded_.isNullAt(row)) {
         if (multiplier_ == 1) {
           result[row] = 0;
         }
-        return;
+        return true;
       }
     }
+
     auto baseIndex = indices[row];
-    uint64_t id = cachedHashes_[baseIndex];
-    if (id == 0) {
-      T value = values[baseIndex];
-
-      if (!success) {
-        // If all were not mappable we just analyze the remaining so we can
-        // decide the hash mode.
-        analyzeValue(value);
-        return;
+    uint64_t& id = cachedHashes_[baseIndex];
```

Comment:
why reference?

Reply:
Sure, because we use `id` for assignment a few lines below.

I can change it to use by-value only.